ratis-test/src/test/java/org/apache/ratis/grpc/TestRetryCacheWithGrpc.java

```diff
@@ -45,6 +47,11 @@
   public void testRetryOnResourceUnavailableException()
       throws InterruptedException, IOException {
     RaftProperties properties = new RaftProperties();
+
+    // make sure leadership check won't affect the test
```

Comment:
test cases invoking `SimpleStateMachine4Testing::blockWriteStateMachineData()` or `DelayLocalExecutionInjection::setDelayMs()` will be affected, which is meet expectation. We need enlarge election timeout to make sure leader won't step down before hearing append entries response from majority followers.

Another case is related to the membership change, especially the one with 9-servers. To avoid flaky behavior in CI environment, we need enlarge election timeout a little bit, say from 300s to 1s, since leader is vulnerable during such situation.